app/src/main/java/org/mozilla/focus/fragment/FragmentDispatcher.kt

```diff
@@ -0,0 +1,134 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.fragment
+
+import android.content.Context
+import android.support.v4.app.FragmentManager
+import android.text.TextUtils
+import org.mozilla.focus.R
+import org.mozilla.focus.session.SessionManager
+import org.mozilla.focus.session.Source
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.telemetry.UrlTextInputLocation
+import org.mozilla.focus.utils.OnUrlEnteredListener
+import org.mozilla.focus.utils.UrlUtils
+import org.mozilla.focus.widget.InlineAutocompleteEditText
+
+class FragmentDispatcher {
+
+    companion object {
+        /**
+         * Loads the given url. If isTextInput is true, there should be no null parameters.
+         */
+        @JvmStatic
+        fun onUrlEnteredInner(urlStr: String, isTextInput: Boolean,
+                              autocompleteResult: InlineAutocompleteEditText.AutocompleteResult?,
+                              inputLocation: UrlTextInputLocation?,
+                              fragmentManager: FragmentManager, sessionManager: SessionManager,
+                              context: Context) {
```

Comment:
nit: I think context should always be the first arg, if it's present. You can fix this quickly in the IDE with the `Change signature` action – it'll update the declaration and the call sites (though maybe it'll mangle line lengths).